docs/reference/hooks.md

```diff
@@ -0,0 +1,154 @@
+ 
+Supported Events:
+1. `pre_commit` - Action runs when the commit occurs, before the commit is finalized.
+2. `pre_merge` - Action runs when the merge occurs, before the merge is finalized.
+ 
+lakeFS `Actions` are handled per repository and cannot be reused.  
+Failure of any `Hook` under any `Action` will result in aborting the lakeFS operation that is taking place.
+
+## Action
+An `Action` is a list of `Hooks` with the same trigger configuration.
+`Hooks` list under an `Action` is ordered and an `Hook` will only be executed if all previous `Hooks` passed.  
+
+### Hook
+An `Hook` is the basic building block of an `Action`. 
+Failure of a single `Hook` will stop the execution of the containing `Action` and fail the `Run`. 
+
+### Action file
+Schema of the Action file:
+
+|Property          |Description                                            |Required |Default Value
+|------------------|-------------------------------------------------------|---------|--------------------------------------|
+|name              |Identify the Action file                               |false    | If missing, filename is used instead 
+|on                |List of events that will trigger the hooks             |true     |
+|on<event>.branches|Glob pattern list of branches that triggers the hooks  |false    | If empty, Action runs on all branches
+|hooks             |List of hooks to be executed                           |true     | 
+|hook.id           |ID of the hook, must be unique within the `Action`     |true     | 
+|hook.type         |Type of the hook (currently only `webhook` supported   |true     | 
+|hook.properties   |Hook's specific configuration                          |true     | 
+
+Example:
+```yaml
+name: Good files check
+description: set of checks to verify that branch is good
+on:
+  pre-commit:
+  pre-merge:
+    branches:
+      - master
+hooks:
+  - id: no_temp
+    type: webhook
+    description: checking no temporary files found
+    properties:
+      url: "https://your.domain.io/webhook?notmp=true?t=1za2PbkZK1bd4prMuTDr6BeEQwWYcX2R"
+  - id: no_freeze
+    type: webhook
+    description: check production is not in dev freeze
+    properties:
+      url: "https://your.domain.io/webhook?nofreeze=true?t=1za2PbkZK1bd4prMuTDr6BeEQwWYcX2R"
+```
+
+## Run
+`Run` is the instantiation of the repo `Action` files when the event occurs.
+lakeFS will fetch, parse and filter the repository `Action` files and start to execute the `Hooks` under each `Action`.
+All executed `Hooks` (each with `hook_run_id`) exists in the context of that `Run` (`run_id`).   
+
+## How to upload Action files
+`Action` files should be uploaded with the prefix `_lakefs_actions/` to the lakeFS repository.
+When an actionable event (see Supported Events above) takes place, lakeFS will read all files with prefix `_lakefs_actions/`
+in the repository branch where the action occurred. 
+A failure to parse an `Action` file will result with the event aborted. 
```

Comment:
Again, not applicable to hooks in general

Reply:
Done.